[google_sign_in]IOS build fails with error: building for iOS Simulator-arm64 but attempting to link with file built for iOS Simulator-x86_64

Please document the fact that this package does not support Apple Silicon. I have now removed support for [google_sign_in] from my project altogether now as it is not supported on Apple Silicon

Here is the flutter Doctor Output plus the error being generated as soon as I add GoogeSignIn package to my yaml file. Everything else functions as expected on. both Android simulator and IOS simulator if I remove this one line from my yaml
google_sign_in: ^5.2.1

[✓] Flutter (Channel stable, 2.5.3, on macOS 12.0.1 21A559 darwin-arm, locale
    en-AU)
    • Flutter version 2.5.3 at /users/alan/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (5 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/alan/Library/Android/sdk/
    • Platform android-Sv2, build-tools 31.0.0
    • ANDROID_HOME = /Users/alan/Library/Android/sdk/
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.1, Build version 13A1030d
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.62.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (3 available)
    • sdk gphone arm64 (mobile) • emulator-5554                        •
      android-arm64  • Android 11 (API 30) (emulator)
    • iPhone 12 mini (mobile)   • 6330F399-7D4D-4F9E-BD5F-DC140A099B27 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • Chrome (web)              • chrome                               •
      web-javascript • Google Chrome 96.0.4664.55
    ! Error: iPhone is not connected. Xcode will continue when iPhone is
      connected. (code -13)


Launching lib/main.dart on iPhone 12 mini in debug mode...
Xcode build done.                                           14.9s
Failed to build iOS app
Error output from Xcode build:
↳
    --- xcodebuild: WARNING: Using the first of multiple matching destinations:
    { platform:iOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-iphonesimulator:placeholder, name:Any iOS Simulator Device }
    { platform:iOS Simulator, id:C18FC93E-5935-4EBA-82E9-92D5A586A815, OS:15.0, name:iPad (9th generation) }
    { platform:iOS Simulator, id:07BEB167-D5D1-47D5-9E5F-31D49F15AFCD, OS:15.0, name:iPad Air (4th generation) }
    { platform:iOS Simulator, id:34D30681-BE68-4717-89FA-1A33C622FCE9, OS:15.0, name:iPad Pro (9.7-inch) }
    { platform:iOS Simulator, id:5D06B5DE-3430-4F77-A954-FF7DD55AE98F, OS:15.0, name:iPad Pro (11-inch) (3rd generation) }
    { platform:iOS Simulator, id:F8EBD39C-2A37-40AA-9FC8-257EE11202C0, OS:15.0, name:iPad Pro (12.9-inch) (5th generation) }
    { platform:iOS Simulator, id:35A48C4A-36A1-4AA2-B790-E005D3CEED32, OS:15.0, name:iPad mini (6th generation) }
    { platform:iOS Simulator, id:054E61CD-362A-4877-97E8-19C536C507BD, OS:15.0, name:iPhone 8 }
    { platform:iOS Simulator, id:80E384C3-6108-4019-BF4E-7927BC4B289C, OS:15.0, name:iPhone 8 Plus }
    { platform:iOS Simulator, id:125AB128-6609-48F2-A3C4-446AD9CE4291, OS:15.0, name:iPhone 11 }
    { platform:iOS Simulator, id:8155E9CE-DA97-4ECC-9016-796EAF6903E6, OS:15.0, name:iPhone 11 Pro }
    { platform:iOS Simulator, id:853EFFB5-0F75-40E9-A8AC-FC512E39CE26, OS:15.0, name:iPhone 11 Pro Max }
    { platform:iOS Simulator, id:481C3AAE-E7C2-4A18-89C3-7174AC85A5D4, OS:15.0, name:iPhone 12 }
    { platform:iOS Simulator, id:F8BC2E6B-ABD2-490B-9F5B-342410650ECD, OS:15.0, name:iPhone 12 Pro }
    { platform:iOS Simulator, id:B70495D7-EF65-4A75-AFD4-427B74D06D8E, OS:15.0, name:iPhone 12 Pro Max }
    { platform:iOS Simulator, id:6330F399-7D4D-4F9E-BD5F-DC140A099B27, OS:15.0, name:iPhone 12 mini }
    { platform:iOS Simulator, id:869D5556-B5D6-4633-A5B5-FB4E6229C727, OS:15.0, name:iPhone 13 }
    { platform:iOS Simulator, id:4ED5A317-0F73-4D28-948C-B5CFAC05FAEA, OS:15.0, name:iPhone 13 Pro }
    { platform:iOS Simulator, id:E0088229-BA38-4625-8624-C03EC325DE89, OS:15.0, name:iPhone 13 Pro Max }
    { platform:iOS Simulator, id:CD599DF8-B75A-4DAF-BE44-F0B39E708668, OS:15.0, name:iPhone 13 mini }
    { platform:iOS Simulator, id:13AEB92B-6B0C-4FDE-95ED-06A155707DAF, OS:15.0, name:iPhone SE (2nd generation) }
    { platform:iOS Simulator, id:3082745A-2815-41C1-8A07-D901F1172375, OS:15.0, name:iPod touch (7th generation) }
    { platform:macOS, arch:arm64, variant:Designed for [iPad,iPhone], id:00006000-001271A63AE2801E }
    { platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Any iOS Device }
    { platform:iOS, id:e50078c82ab6bea2add8c4e3f324d2150f6cb08e, name:iPhone }
    ** BUILD FAILED **
Xcode's output:
↳
    ld: warning: ignoring file /Users/alan/Documents/flutter_apps/mindful/build/ios/Debug-iphonesimulator/google_sign_in/google_sign_in.framework/google_sign_in, building for iOS Simulator-arm64 but attempting to link with file built for iOS Simulator-x86_64
    gRPC-C++-gRPCCertificates-Cpp-fpzkavsonpbsiudzcphnjvrehylu
    ld: in /Users/alan/Documents/flutter_apps/mindful/ios/Pods/GoogleSignIn/Frameworks/GoogleSignIn.framework/GoogleSignIn(GIDEMMErrorHandler_3a47e13d8ca81b41e9cdb7ef5468004a.o), building for iOS Simulator, but linking in object file built for iOS, file '/Users/alan/Documents/flutter_apps/mindful/ios/Pods/GoogleSignIn/Frameworks/GoogleSignIn.framework/GoogleSignIn' for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    gRPC-C++-gRPCCertificates-Cpp-fpzkavsonpbsiudzcphnjvrehylu
    note: Using new build system
    note: Planning
    note: Build preparation complete
    note: Building targets in parallel
    /Users/alan/Documents/flutter_apps/mindful/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.0.99. (in target 'abseil' from project 'Pods')
    /Users/alan/Documents/flutter_apps/mindful/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.0.99. (in target 'FMDB' from project 'Pods')
    /Users/alan/Documents/flutter_apps/mindful/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.0.99. (in target 'BoringSSL-GRPC' from project 'Pods')
    /Users/alan/Documents/flutter_apps/mindful/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.0.99. (in target 'AppAuth' from project 'Pods')
    /Users/alan/Documents/flutter_apps/mindful/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.0.99. (in target 'gRPC-Core' from project 'Pods')
    /Users/alan/Documents/flutter_apps/mindful/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.0.99. (in target 'gRPC-C++' from project 'Pods')
    /Users/alan/Documents/flutter_apps/mindful/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.0.99. (in target 'GoogleSignIn' from project 'Pods')
    /Users/alan/Documents/flutter_apps/mindful/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.0.99. (in target 'leveldb-library' from project 'Pods')
    /Users/alan/Documents/flutter_apps/mindful/ios/Pods/Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.0.99. (in target 'gRPC-C++-gRPCCertificates-Cpp' from project 'Pods')
Could not build the application for the simulator.
Error launching application on iPhone 12 mini.
Exited (sigterm)

--- a/packages/flutter_tools/lib/src/macos/cocoapods.dart
+++ b/packages/flutter_tools/lib/src/macos/cocoapods.dart
@@ -360,8 +360,7 @@ class CocoaPods {
         '  pod repo update\n',
         emphasis: true,
       );
-    } else if (stdout.contains('Init_ffi_c') &&
-        stdout.contains('symbol not found') &&
+    } else if (stdout.contains('ffi_c.bundle') && stdout.contains('LoadError') &&
         _operatingSystemUtils.hostPlatform == HostPlatform.darwin_arm) {
       // https://github.com/flutter/flutter/issues/70796
       UsageEvent(

--- a/packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart
+++ b/packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart
@@ -491,46 +491,52 @@ Note: as of CocoaPods 1.0, `pod repo update` does not happen on `pod install` by
       );
     });
 
-    testUsingContext('ffi failure on ARM macOS prompts gem install', () async {
-      final FlutterProject projectUnderTest = setupProjectUnderTest();
-      pretendPodIsInstalled();
-      pretendPodVersionIs('1.10.0');
-      fileSystem.file(fileSystem.path.join('project', 'ios', 'Podfile'))
-        ..createSync()
-        ..writeAsStringSync('Existing Podfile');
+    final Map<String, String> possibleErrors = <String, String>{
+      'symbol not found': 'LoadError - dlsym(0x7fbbeb6837d0, Init_ffi_c): symbol not found - /Library/Ruby/Gems/2.6.0/gems/ffi-1.13.1/lib/ffi_c.bundle',
+      'incompatible architecture': "LoadError - (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/lib/ffi_c.bundle' (no such file) - /Library/Ruby/Gems/2.6.0/gems/ffi-1.15.4/lib/ffi_c.bundle",
+    };
+    possibleErrors.forEach((String errorName, String cocoaPodsError) {
+      testUsingContext('ffi $errorName failure on ARM macOS prompts gem install', () async {
+        final FlutterProject projectUnderTest = setupProjectUnderTest();
+        pretendPodIsInstalled();
+        pretendPodVersionIs('1.10.0');
+        fileSystem.file(fileSystem.path.join('project', 'ios', 'Podfile'))
+          ..createSync()
+          ..writeAsStringSync('Existing Podfile');
 
-      fakeProcessManager.addCommands(<FakeCommand>[
-        const FakeCommand(
-          command: <String>['pod', 'install', '--verbose'],
-          workingDirectory: 'project/ios',
-          environment: <String, String>{
-            'COCOAPODS_DISABLE_STATS': 'true',
-            'LANG': 'en_US.UTF-8',
-          },
-          exitCode: 1,
-          stdout: 'LoadError - dlsym(0x7fbbeb6837d0, Init_ffi_c): symbol not found - /Library/Ruby/Gems/2.6.0/gems/ffi-1.13.1/lib/ffi_c.bundle',
-        ),
-        const FakeCommand(
-          command: <String>['which', 'sysctl'],
-        ),
-        const FakeCommand(
-          command: <String>['sysctl', 'hw.optional.arm64'],
-          stdout: 'hw.optional.arm64: 1',
-        ),
-      ]);
+        fakeProcessManager.addCommands(<FakeCommand>[
+          FakeCommand(
+            command: const <String>['pod', 'install', '--verbose'],
+            workingDirectory: 'project/ios',
+            environment: const <String, String>{
+              'COCOAPODS_DISABLE_STATS': 'true',
+              'LANG': 'en_US.UTF-8',
+            },
+            exitCode: 1,
+            stdout: cocoaPodsError,
+          ),
+          const FakeCommand(
+            command: <String>['which', 'sysctl'],
+          ),
+          const FakeCommand(
+            command: <String>['sysctl', 'hw.optional.arm64'],
+            stdout: 'hw.optional.arm64: 1',
+          ),
+        ]);
 
-      await expectToolExitLater(
-        cocoaPodsUnderTest.processPods(
-          xcodeProject: projectUnderTest.ios,
-          buildMode: BuildMode.debug,
-        ),
-        equals('Error running pod install'),
-      );
-      expect(
-        logger.errorText,
-        contains('set up CocoaPods for ARM macOS'),
-      );
-      expect(usage.events, contains(const TestUsageEvent('pod-install-failure', 'arm-ffi')));
+        await expectToolExitLater(
+          cocoaPodsUnderTest.processPods(
+            xcodeProject: projectUnderTest.ios,
+            buildMode: BuildMode.debug,
+          ),
+          equals('Error running pod install'),
+        );
+        expect(
+          logger.errorText,
+          contains('set up CocoaPods for ARM macOS'),
+        );
+        expect(usage.events, contains(const TestUsageEvent('pod-install-failure', 'arm-ffi')));
+      });
     });
 
     testUsingContext('ffi failure on x86 macOS does not prompt gem install', () async {

